### PR TITLE
Homework w01 d04 - i/O & Enumeration

### PR DESCRIPTION
**Pull request by mkikegawa**
{
  "comfortability": 4,
  "completeness": 4,
  "Comments: to be implemented - write modified grocery list to new file
                       bugs - author info not formatting correctly"
}

### CODE REVIEW

**Comment by mkikegawa** [comment]
please ignore talkative_robot-v3.rb i forgot to delete it

**Comment by rgpass** [comment]
Not sure why this was closed.

File: talkative_robot.rb [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+# author = { :name=>"Martin", :age=>10, :gender=>"M", :shopping_cart=>[], :author=> "yes", :description=>"boy", :initial=>"M", :destination=>"n" }
+# user   = { :name=>"Ernest", :age=>10, :gender=>"M", :shopping_cart=>[], :grocery_list => ["apples", "oranges", "peaches", "kale"], :author=> nil, :description=>"boy", :initial=>"M", :destination=>"n" }
+# people = [author, user]
+###  remove this block when done testing -- end here
+
+
+
+def number_of_linefeeds(num)
+  num.times {puts}
+end
```
Comment (rgpass) [inline talkative_robot.rb:18]: Creative!

File: talkative_robot.rb [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+def print_is_user_ernest_message(user)
+  puts user[:name] == "Ernest" ? "I knew it was you!" : "Sorry, you're not Ernest."
+end
+
+
+def get_user_name
+	puts "What's your name?"
+ 	gets.chomp.capitalize
```
Comment (rgpass) [inline talkative_robot.rb:27]: Tabbing looks weird. May just be GitHub. If it looks right on your screen, ignore this comment.

File: talkative_robot.rb [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+ 	gets.chomp.upcase
+end
+
+def young?(user)
+  user[:age] <= 16
+end 
+
+def old?(user)
+	user[:age] >= 100
+end
```
Comment (rgpass) [inline talkative_robot.rb:46]: Yay!

File: talkative_robot.rb [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+
+def when_is_user_is_75(user)
+  delta = (75 - user[:age]).abs
+  case user[:age]
+ 	when 0..74
+ 		puts "You'll be 75 in #{delta} years."
+ 	when 75
+ 		puts "You're 75!!!!"
+  when 76..150
```
Comment (rgpass) [inline talkative_robot.rb:55]: Tabbing

File: talkative_robot.rb [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+def load_grocery_list(file)
+  grocery_list = IO.read(file).split("\n")
+end
+
+def get_user_info
+  name   = get_user_name
+  age    = get_user_age
+  gender = get_user_gender 
+  shopping_cart = []
+  {name: name, age: age, gender: gender, shopping_cart: shopping_cart}
```
Comment (rgpass) [inline talkative_robot.rb:95]: Add a space after the first `{` and a space before the last `}`

**Comment by rgpass** [comment]
Either I missed it or there's no use of `select` to select the author. Do this then I'll give the :+1: -- everything else looks awesome and clean!

**Pull request merged**